com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/validation/ScopeJavaValidator.java

```diff
@@ -277,6 +281,45 @@ public void checkScopeRuleOverriding(final ScopeModel context) {
     }
   }
 
+  /**
+   * Verifies that a matching default name function exists for scope expressions without explicit name functions.
+   *
+   * @param expr
+   *          scope expression to check
+   */
+  @Check
+  public void checkNameFunctionExists(final SimpleScopeExpression expr) {
+    if (expr.getNaming() != null && !expr.getNaming().getNames().isEmpty()) {
+      return;
+    }
+    ScopeDefinition def = EObjectUtil.eContainer(expr, ScopeDefinition.class);
+    ScopeModel model = EObjectUtil.eContainer(expr, ScopeModel.class);
+    if (def != null && model != null) {
+      EClass scopeType = getType(def);
+      NamingSection namingSection = model.getNaming();
+      if (namingSection != null) {
+        for (NamingDefinition naming : namingSection.getNamings()) {
+          if (naming.getType() != null && isLeftMostSuperType(naming.getType(), scopeType)) {

```

Comment:
mm, why only the left-most supertype?

Reply:
Because that matches what the scope name provider also does.

Reply:
hmm, ok